asv_bench/benchmarks/utils/common.py

```diff
@@ -498,21 +498,26 @@ def execute(
         if ASV_USE_STORAGE_FORMAT == "omnisci":
             df._query_compiler._modin_frame._execute()
             return
-        partitions = df._query_compiler._modin_frame._partitions
+        partitions = df._query_compiler._modin_frame._partitions.flatten()
+        if hasattr(partitions[0], "wait"):
+            all(map(lambda partition: partition.wait(), partitions))
+            return
```

Comment:
Is the `return` statement here for synchronization purposes across the different partitions?

Reply:
This statement is here, since the previous line is enough to wait for the end of all operations. However, in order to run `ASV` benchmarks for previous versions of `Modin`, we must leave the code that comes after this statement.

Although I'm not sure that answered your question.